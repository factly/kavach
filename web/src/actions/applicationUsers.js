import axios from 'axios';
import {
  ADD_APPLICATION_USER,
  ADD_APPLICATION_USERS,
  ADD_APPLICATION_USERS_REQUEST,
  SET_APPLICATION_USERS_LOADING,
  RESET_APPLICATION_USERS,
  APPLICATION_USERS_API,
  ADD_USER_IDS,
} from '../constants/applicationUser';
import { getIds } from '../utils/objects';
import { addMediaList } from './media';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { loadingUsers, stopUsersLoading } from './users';

export const getApplications = (appID) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .get(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          appID +
          '/users',
      )
      .then((response) => {
        dispatch(
          addApplicationList(
            response.data.users.map((user) => {
              return user;
            }),
          ),
        );
        dispatch(
          addApplicationsRequest({
            data: response.data.users.map((item) => item.id),
            application_id: response.data.application.id,
            total: response.data.total,
          }),
        );
        dispatch(stopApplicationLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const getApplication = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .get(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          id +
          '/users',
      )
      .then((response) => {
        if (response.data.medium) dispatch(addMediaList([response.data.medium]));
        dispatch(getApplicationByID({ ...response.data, medium: response.data.medium?.id }));
        dispatch(stopApplicationLoading());
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const addApplicationUser = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .post(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          data.application_id +
          '/users',
        data,
      )
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application User Added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const updateApplication = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .put(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          data.id,
        data,
      )
      .then((response) => {
        if (response.data.medium) dispatch(addMediaList([response.data.medium]));
        dispatch(getApplicationByID({ ...response.data, medium: response.data.medium?.id }));
        dispatch(stopApplicationLoading());
        dispatch(addSuccessNotification('Application Updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const deleteApplication = (id, appID) => {
  return (dispatch, getState) => {
    dispatch(loadingApplications());
    return axios
      .delete(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          id +
          '/users/' +
          appID,
      )
      .then(() => {
        dispatch(resetApplications());
        dispatch(addSuccessNotification('Application User Deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const getApplicationUsers = (appID) => {
  return (dispatch, getState) => {
    dispatch(loadingUsers());
    return axios
      .get(
        APPLICATION_USERS_API +
          '/' +
          getState().organisations.selected +
          '/applications/' +
          appID +
          '/users',
      )
      .then((res) => {
        dispatch(addUserIds(getIds(res.data.users), appID));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopUsersLoading());
      });
  };
};

export const addApplications = (applications) => {
  return (dispatch) => {
    dispatch(
      addMediaList(
        applications
          .filter((application) => application.medium)
          .map((application) => application.medium),
      ),
    );
    dispatch(
      addApplicationList(
        applications.map((application) => {
          return { ...application, medium: application.medium?.id };
        }),
      ),
    );
  };
};

export const loadingApplications = () => ({
  type: SET_APPLICATION_USERS_LOADING,
  payload: true,
});

export const stopApplicationLoading = () => ({
  type: SET_APPLICATION_USERS_LOADING,
  payload: false,
});

export const getApplicationByID = (data) => ({
  type: ADD_APPLICATION_USER,
  payload: data,
});

export const addApplicationList = (data) => ({
  type: ADD_APPLICATION_USERS,
  payload: data,
});

export const addApplicationsRequest = (data) => ({
  type: ADD_APPLICATION_USERS_REQUEST,
  payload: data,
});

const addUserIds = (data, appID) => ({
  type: ADD_USER_IDS,
  payload: {
    id: appID,
    data: data,
  },
});
export const resetApplications = () => ({
  type: RESET_APPLICATION_USERS,
});
