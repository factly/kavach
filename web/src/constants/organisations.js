//Actions
export const ADD_ORGANISATION = 'ADD_ORGANISATION';
export const ADD_ORGANISATIONS = 'ADD_ORGANISATIONS';
export const RESET_ORGANISATIONS = 'RESET_ORGANISATIONS';
export const SET_ORGANISATIONS_LOADING = 'SET_ORGANISATIONS_LOADING';
export const SET_SELECTED_ORGANISATION = 'SET_SELECTED_ORGANISATION';
export const ADD_ORGANISATION_USERS = 'ADD_ORGANISATION_USERS';
export const ADD_APPLICATION_IDS = 'ADD_APPLICATION_IDS';
export const ADD_ORGANISATION_ROLE = 'ADD_ORGANISATION_ROLE';
//API
export const ORGANISATIONS_API = '/organisations';
