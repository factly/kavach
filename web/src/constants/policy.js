// API
export const POLICY_API = '/policy';

//SPACE POLICY
export const ADD_ORGANISATION_POLICY = 'ADD_ORGANISATION_POLICY';
export const DELETE_ORGANISATION_POLICY = 'DELETE_ORGANISATION_POLICY';

//APPLICATION POLICY
export const ADD_APPLICATION_POLICY = 'ADD_APPLICATION_POLICY';
export const DELETE_APPLICATION_POLICY = 'DELETE_APPLICATION_POLICY';

//SPACE POLICY
export const ADD_SPACE_POLICY = 'ADD_SPACE_POLICY';
export const DELETE_SPACE_POLICY = 'DELETE_SPACE_POLICY';
