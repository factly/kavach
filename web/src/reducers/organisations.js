import {
  ADD_ORGANISATION,
  ADD_ORGANISATIONS,
  SET_ORGANISATIONS_LOADING,
  RESET_ORGANISATIONS,
  SET_SELECTED_ORGANISATION,
} from '../constants/organisations';

const initialState = {
  ids: [],
  details: {},
  loading: true,
  selected: 0,
};

export default function tagsReducer(state = initialState, action = {}) {
  switch (action.type) {
    case RESET_ORGANISATIONS:
      return {
        ...state,
        ids: [],
        details: {},
        loading: true,
        selected: 0,
      };
    case SET_ORGANISATIONS_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    case ADD_ORGANISATIONS:
      return {
        ...state,
        ids: action.payload.map((item) => item.id),
        details: action.payload.reduce((obj, item) => Object.assign(obj, { [item.id]: item }), {}),
        selected: action.payload[0].id,
      };
    case ADD_ORGANISATION:
      return {
        ...state,
        ids: state.ids.includes(action.payload.id)
          ? state.ids
          : state.ids.concat([action.payload.id]),
        details: {
          ...state.details,
          [action.payload.id]: action.payload,
        },
      };
    case SET_SELECTED_ORGANISATION:
      return {
        ...state,
        selected: action.payload,
      };
    default:
      return state;
  }
}
