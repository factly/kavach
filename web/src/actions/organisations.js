import axios from 'axios';
import {
  ADD_ORGANISATION,
  ADD_ORGANISATIONS,
  SET_ORGANISATIONS_LOADING,
  RESET_ORGANISATIONS,
  ORGANISATIONS_API,
  SET_SELECTED_ORGANISATION,
} from '../constants/organisations';
import { ADD_ORGANISATION_IDS } from '../constants/profile';
import { buildObjectOfItems, deleteKeys, getIds, getValues } from '../utils/objects';
import { addApplicationList } from './application';
import { addMedia, addMediaList } from './media';
import { addErrorNotification, addSuccessNotification } from './notifications';
import { addUsersList } from './users';

export const getOrganisations = () => {
  return (dispatch, getState) => {
    dispatch(loadingOrganisations());
    return axios
      .get(ORGANISATIONS_API + '/my')
      .then((response) => {
        dispatch(addOrganisationsList(response.data));
        dispatch(addOrganisationIds(getIds(response.data)));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopOrganisationsLoading());
      });
  };
};

export const getOrganisation = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganisations());
    return axios
      .get(ORGANISATIONS_API + '/' + id)
      .then((response) => {
        if (response.featured_medium_id) {
          addMedia(response.data.medium);
        }
        let users = [];
        response.data.roles = {};
        response.data.organisation_users.map((item) => {
          users.push(item.user);
          response.data.roles[item.user.id] = item.role;
          return null;
        });
        response.data.role = response.data.permission.role;
        deleteKeys([response.data], ['permission', 'organisation_users']);
        response.data.applications = getIds(response.data.applications);
        response.data.users = getIds(users);
        dispatch(getOrganisationByID(response.data));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(()=>{
        dispatch(stopOrganisationsLoading());
      })
      ;
  };
};

export const addOrganisation = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganisations());
    return axios
      .post(ORGANISATIONS_API, data)
      .then((response) => {
        dispatch(getOrganisationByID(response.data));
        dispatch(setSelectedOrganisation(response.data.id));
        dispatch(addSuccessNotification('Organisation added'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      })
      .finally(() => {
        dispatch(stopOrganisationsLoading());
      });
  };
};

export const updateOrganisation = (data) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganisations());
    return axios
      .put(ORGANISATIONS_API + '/' + data.id, data)
      .then((response) => {
        dispatch(getOrganisationByID(response.data));
        dispatch(stopOrganisationsLoading());
        dispatch(addSuccessNotification('Organisation Updated'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const deleteOrganisation = (id) => {
  return (dispatch, getState) => {
    dispatch(loadingOrganisations());
    return axios
      .delete(ORGANISATIONS_API + '/' + id)
      .then(() => {
        dispatch(resetOrganisations());
        dispatch(addSuccessNotification('Organisation Deleted'));
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const setSelectedOrganisation = (id) => ({
  type: SET_SELECTED_ORGANISATION,
  payload: id,
});

export const loadingOrganisations = () => ({
  type: SET_ORGANISATIONS_LOADING,
  payload: true,
});

export const stopOrganisationsLoading = () => ({
  type: SET_ORGANISATIONS_LOADING,
  payload: false,
});

export const getOrganisationByID = (data) => ({
  type: ADD_ORGANISATION,
  payload: data,
});

export const addOrganisationsList = (data, id) => (dispatch) => {
  const medium = getValues(data, 'medium');
  dispatch(addMediaList(medium));
  const applications = getValues(data, 'applications');
  dispatch(addApplicationList(applications));
  data.forEach((organisation) => {
    let users = [];
    organisation.roles = {};
    organisation.organisation_users.map((item) => {
      users.push(item.user);
      if (item.user.id === id) {
        organisation.role = item.role;
      }
      organisation.roles[item.user.id] = item.role;
      return null;
    });
    dispatch(addUsersList(users));
    organisation.users = getIds(users);
    organisation.applications = getIds(organisation.applications);
    deleteKeys([organisation], ['organisation_users', 'medium']);
  });
  const ids = getIds(data);
  dispatch({
    type: ADD_ORGANISATIONS,
    payload: {
      data: buildObjectOfItems(data),
      ids: ids,
    },
  });
};

export const resetOrganisations = () => ({
  type: RESET_ORGANISATIONS,
});

export const addOrganisationIds = (ids) => (dispatch) => {
  dispatch({
    type: ADD_ORGANISATION_IDS,
    payload: ids,
  });
};
