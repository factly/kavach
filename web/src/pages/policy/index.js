import React from 'react';
import { Tabs, Form, Collapse } from 'antd';
import PolicyList from './components/PolicyList';
import PolicyForm from './components/PolicyForm';
import { createApplicationPolicy, createOrganisationPolicy, createSpacePolicy, getApplicationPolicy, getOrganisationPolicy, getSpacePolicy } from '../../actions/policy';
import { useDispatch } from 'react-redux';

function Policy() {
  const [form] = Form.useForm();
  const dispatch = useDispatch();
  const onSubmit = (value, type) => {  
      switch (type) {
        case 'organisation':
          dispatch(createOrganisationPolicy(value)).then(() => dispatch(getOrganisationPolicy()));
          break;
        case 'application':
          dispatch(createApplicationPolicy(value.application, value)).then(() =>
            getApplicationPolicy(value.application),
          );
          break;
        case 'space':
          dispatch(createSpacePolicy(value.application, value.space, value)).then(() =>
            dispatch(getSpacePolicy(value.application, value.space)),
          );
          break;
        default:
          return;
      }
      form.resetFields();
    };

  return (
    <Collapse>
      <Collapse.Panel header="Create Policy">
        <Tabs centered={true}>
          <Tabs.TabPane tab="Organisation Policy" key="1">
            <PolicyForm type="organisation" onSubmit={onSubmit} form={form}/>
          </Tabs.TabPane>
          <Tabs.TabPane tab="Application " key="2">
            <PolicyForm type="application" onSubmit={onSubmit} form={form}/>
          </Tabs.TabPane>
          <Tabs.TabPane tab="Space Policy" key="3">
            <PolicyForm type="space" onSubmit={onSubmit} form={form}/>
          </Tabs.TabPane>
        </Tabs>
      </Collapse.Panel>
      <Collapse.Panel header="View Policy">
        <Tabs centered={true}>
          <Tabs.TabPane tab="Organisation Policy" key="1">
            <PolicyList type="organisation" />
          </Tabs.TabPane>
          <Tabs.TabPane tab="Application Policy" key="2">
            <PolicyList type="application" />
          </Tabs.TabPane>
          <Tabs.TabPane tab="Space Policy" key="3">
            <PolicyList type="space" />
          </Tabs.TabPane>
        </Tabs>
      </Collapse.Panel>
    </Collapse>
  );
}

export default Policy;
