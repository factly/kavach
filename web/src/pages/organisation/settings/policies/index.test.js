import React from 'react';
import thunk from 'redux-thunk';
import { useDispatch } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { useHistory, useParams } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import { getOrganisation } from '../../../../actions/organisations';
import { getOrganisationPolicy } from '../../../../actions/policy';

import OrganisationPolicies from './index';

import '../../../../matchMedia.mock.js';
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
  useParams: jest.fn(() => ({
    orgID: 1,
  })),
}));

jest.mock('../../../../actions/organisations', () => ({
  getOrganisation: jest.fn(),
}));

jest.mock('../../../../actions/policy', () => ({
  getOrganisationPolicy: jest.fn(),
}));

jest.mock('./components/PolicyList', () => {
  return {
    __esModule: true,
    default: (props) => {
      return <div id="policyList" onClick={props.onChange} />;
    },
  };
});

let state = {
  organisations: {
    details: {
      1: {
        id: 1,
        name: 'test',
        roleIDs: [1, 2, 3],
      },
    },
    loading: false,
    selected: 1,
  },
  profile: {
    roles: {
      1: 'owner',
    },
    loading: false,
  },
};

describe('OrganisationPolicies component', () => {
  let store;
  const mockedDispatch = jest.fn();
  mockedDispatch.mockReturnValue(Promise.resolve());
  useDispatch.mockReturnValue(mockedDispatch);
  store = mockStore({});
  store.dispatch = jest.fn(() => ({}));
  const push = jest.fn();
  useHistory.mockReturnValue({ push });
  const description = 'should render the OrganisationPolicies component';
  describe('snapshot testing', () => {
    it(description + ' when loadingorg', () => {
      store = mockStore({
        ...state,
        organisations: {
          ...state.organisations,
          loading: true,
        },
        profile: {
          ...state.profile,
          loading: true,
        },
      });
      const tree = mount(
        <Provider store={store}>
          <Router>
            <OrganisationPolicies />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });

    it(description + ' when role is not owner', () => {
      store = mockStore({
        ...state,
        profile: {
          ...state.profile,
          roles: {
            1: 'member',
          },
        },
      });
      const tree = mount(
        <Provider store={store}>
          <Router>
            <OrganisationPolicies />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
    it(description + ' when role is owner', () => {
      store = mockStore({
        ...state,
      });
      const tree = mount(
        <Provider store={store}>
          <Router>
            <OrganisationPolicies />
          </Router>
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe('function testing', () => {
    it('should call getOrganisation and getOrganisationPolicy om render', () => {
      store = mockStore({
        ...state,
      });
      mount(
        <Provider store={store}>
          <Router>
            <OrganisationPolicies />
          </Router>
        </Provider>,
      );
      expect(getOrganisation).toHaveBeenCalledWith(1);
      expect(getOrganisationPolicy).toHaveBeenCalledWith();
    });
  });
});
