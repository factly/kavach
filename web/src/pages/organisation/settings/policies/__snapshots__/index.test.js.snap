// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationPolicies component snapshot testing should render the OrganisationPolicies component when loadingorg 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <OrganisationPolicies>
        <Space
          direction="vertical"
        >
          <div
            className="ant-space ant-space-vertical"
            style={Object {}}
          >
            <Item
              className="ant-space-item"
              direction="vertical"
              index={0}
              key="1"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <Link
                  key="1"
                  to="/organisation"
                >
                  <LinkAnchor
                    href="/organisation"
                    navigate={[Function]}
                  >
                    <a
                      href="/organisation"
                      onClick={[Function]}
                    >
                      <Button
                        type="primary"
                      >
                        <Wave
                          disabled={false}
                        >
                          <button
                            className="ant-btn ant-btn-primary"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            <LoadingIcon
                              existIcon={false}
                              loading={false}
                              prefixCls="ant-btn"
                            >
                              <CSSMotion
                                motionName="ant-btn-loading-icon-motion"
                                onAppearActive={[Function]}
                                onAppearStart={[Function]}
                                onEnterActive={[Function]}
                                onEnterStart={[Function]}
                                onLeaveActive={[Function]}
                                onLeaveStart={[Function]}
                                removeOnLeave={true}
                                visible={false}
                              >
                                <DomWrapper />
                              </CSSMotion>
                            </LoadingIcon>
                            <span
                              key=".0"
                            >
                               Back to Settings 
                            </span>
                          </button>
                        </Wave>
                      </Button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={1}
              key="ant-space-item-1"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <h2
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  Policies in 
                </h2>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={2}
              key="ant-space-item-2"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={Object {}}
              >
                <Skeleton>
                  <div
                    className="ant-skeleton"
                  >
                    <div
                      className="ant-skeleton-content"
                    >
                      <Title
                        prefixCls="ant-skeleton-title"
                        width="38%"
                      >
                        <h3
                          className="ant-skeleton-title"
                          style={
                            Object {
                              "width": "38%",
                            }
                          }
                        />
                      </Title>
                      <Paragraph
                        prefixCls="ant-skeleton-paragraph"
                        rows={3}
                        width="61%"
                      >
                        <ul
                          className="ant-skeleton-paragraph"
                        >
                          <li
                            key="0"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          />
                          <li
                            key="1"
                            style={
                              Object {
                                "width": undefined,
                              }
                            }
                          />
                          <li
                            key="2"
                            style={
                              Object {
                                "width": "61%",
                              }
                            }
                          />
                        </ul>
                      </Paragraph>
                    </div>
                  </div>
                </Skeleton>
              </div>
            </Item>
          </div>
        </Space>
      </OrganisationPolicies>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`OrganisationPolicies component snapshot testing should render the OrganisationPolicies component when role is not owner 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <OrganisationPolicies>
        <Space
          direction="vertical"
        >
          <div
            className="ant-space ant-space-vertical"
            style={Object {}}
          >
            <Item
              className="ant-space-item"
              direction="vertical"
              index={0}
              key="1"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <Link
                  key="1"
                  to="/organisation"
                >
                  <LinkAnchor
                    href="/organisation"
                    navigate={[Function]}
                  >
                    <a
                      href="/organisation"
                      onClick={[Function]}
                    >
                      <Button
                        type="primary"
                      >
                        <Wave
                          disabled={false}
                        >
                          <button
                            className="ant-btn ant-btn-primary"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            <LoadingIcon
                              existIcon={false}
                              loading={false}
                              prefixCls="ant-btn"
                            >
                              <CSSMotion
                                motionName="ant-btn-loading-icon-motion"
                                onAppearActive={[Function]}
                                onAppearStart={[Function]}
                                onEnterActive={[Function]}
                                onEnterStart={[Function]}
                                onLeaveActive={[Function]}
                                onLeaveStart={[Function]}
                                removeOnLeave={true}
                                visible={false}
                              >
                                <DomWrapper />
                              </CSSMotion>
                            </LoadingIcon>
                            <span
                              key=".0"
                            >
                               Back to Settings 
                            </span>
                          </button>
                        </Wave>
                      </Button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={1}
              key="ant-space-item-1"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <h2
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  Policies in 
                </h2>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={2}
              key="ant-space-item-2"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={Object {}}
              >
                <Space
                  direction="vertical"
                >
                  <div
                    className="ant-space ant-space-vertical"
                    style={Object {}}
                  >
                    <Item
                      className="ant-space-item"
                      direction="vertical"
                      index={0}
                      key="ant-space-item-0"
                      marginDirection="marginRight"
                      wrap={false}
                    />
                    <Item
                      className="ant-space-item"
                      direction="vertical"
                      index={1}
                      key="policyList-1"
                      marginDirection="marginRight"
                      wrap={false}
                    >
                      <div
                        className="ant-space-item"
                        style={Object {}}
                      >
                        <default
                          key="policyList-1"
                          orgID={1}
                          role="member"
                        >
                          <div
                            id="policyList"
                          />
                        </default>
                      </div>
                    </Item>
                  </div>
                </Space>
              </div>
            </Item>
          </div>
        </Space>
      </OrganisationPolicies>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`OrganisationPolicies component snapshot testing should render the OrganisationPolicies component when role is owner 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <OrganisationPolicies>
        <Space
          direction="vertical"
        >
          <div
            className="ant-space ant-space-vertical"
            style={Object {}}
          >
            <Item
              className="ant-space-item"
              direction="vertical"
              index={0}
              key="1"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <Link
                  key="1"
                  to="/organisation"
                >
                  <LinkAnchor
                    href="/organisation"
                    navigate={[Function]}
                  >
                    <a
                      href="/organisation"
                      onClick={[Function]}
                    >
                      <Button
                        type="primary"
                      >
                        <Wave
                          disabled={false}
                        >
                          <button
                            className="ant-btn ant-btn-primary"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            <LoadingIcon
                              existIcon={false}
                              loading={false}
                              prefixCls="ant-btn"
                            >
                              <CSSMotion
                                motionName="ant-btn-loading-icon-motion"
                                onAppearActive={[Function]}
                                onAppearStart={[Function]}
                                onEnterActive={[Function]}
                                onEnterStart={[Function]}
                                onLeaveActive={[Function]}
                                onLeaveStart={[Function]}
                                removeOnLeave={true}
                                visible={false}
                              >
                                <DomWrapper />
                              </CSSMotion>
                            </LoadingIcon>
                            <span
                              key=".0"
                            >
                               Back to Settings 
                            </span>
                          </button>
                        </Wave>
                      </Button>
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={1}
              key="ant-space-item-1"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={
                  Object {
                    "marginBottom": 8,
                  }
                }
              >
                <h2
                  style={
                    Object {
                      "marginTop": 20,
                    }
                  }
                >
                  Policies in 
                </h2>
              </div>
            </Item>
            <Item
              className="ant-space-item"
              direction="vertical"
              index={2}
              key="ant-space-item-2"
              marginDirection="marginRight"
              wrap={false}
            >
              <div
                className="ant-space-item"
                style={Object {}}
              >
                <Space
                  direction="vertical"
                >
                  <div
                    className="ant-space ant-space-vertical"
                    style={Object {}}
                  >
                    <Item
                      className="ant-space-item"
                      direction="vertical"
                      index={0}
                      key="ant-space-item-0"
                      marginDirection="marginRight"
                      wrap={false}
                    >
                      <div
                        className="ant-space-item"
                        style={
                          Object {
                            "marginBottom": 8,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "display": "flex",
                              "justifyContent": "end",
                            }
                          }
                        >
                          <Link
                            key="1"
                            to="/organisation/1/settings/policies/create"
                          >
                            <LinkAnchor
                              href="/organisation/1/settings/policies/create"
                              navigate={[Function]}
                            >
                              <a
                                href="/organisation/1/settings/policies/create"
                                onClick={[Function]}
                              >
                                <Button
                                  type="primary"
                                >
                                  <Wave
                                    disabled={false}
                                  >
                                    <button
                                      className="ant-btn ant-btn-primary"
                                      disabled={false}
                                      onClick={[Function]}
                                      type="button"
                                    >
                                      <LoadingIcon
                                        existIcon={false}
                                        loading={false}
                                        prefixCls="ant-btn"
                                      >
                                        <CSSMotion
                                          motionName="ant-btn-loading-icon-motion"
                                          onAppearActive={[Function]}
                                          onAppearStart={[Function]}
                                          onEnterActive={[Function]}
                                          onEnterStart={[Function]}
                                          onLeaveActive={[Function]}
                                          onLeaveStart={[Function]}
                                          removeOnLeave={true}
                                          visible={false}
                                        >
                                          <DomWrapper />
                                        </CSSMotion>
                                      </LoadingIcon>
                                      <span
                                        key=".0"
                                      >
                                         Create New Policies 
                                      </span>
                                    </button>
                                  </Wave>
                                </Button>
                              </a>
                            </LinkAnchor>
                          </Link>
                        </div>
                      </div>
                    </Item>
                    <Item
                      className="ant-space-item"
                      direction="vertical"
                      index={1}
                      key="policyList-1"
                      marginDirection="marginRight"
                      wrap={false}
                    >
                      <div
                        className="ant-space-item"
                        style={Object {}}
                      >
                        <default
                          key="policyList-1"
                          orgID={1}
                          role="owner"
                        >
                          <div
                            id="policyList"
                          />
                        </default>
                      </div>
                    </Item>
                  </div>
                </Space>
              </div>
            </Item>
          </div>
        </Space>
      </OrganisationPolicies>
    </Router>
  </BrowserRouter>
</Provider>
`;
